### PR TITLE
fix: conditionally set CRAN repo

### LINKED ISSUE #2: Make R languageserver installation respect alternative/user-defined `options("repos")`
@williamboman - hope this is the right place to address this?
Problem description - "steps to reproduce" (but only if you have a somewhat strict firewall in place, if there is not tightened security around, this will not be an issue, prob just a feature?).
Start Neovim
LspInstall (or MasonInstall) is expected to be available.
Not mandatory: open a .R file
Hit ":" (to enter Command mode) type `LspInstall r_language_server` (or MasonInstall)
Watch it use "http://cran.rstudio.com/" which in my case fails setting up the lang server for neovim b/c some packages (like callr, rprojroot, etc.) will be blocked from downloading from this mirror but not an "internal" one defined in my ~/.Rprofile.
Found that to be caused by line 8 in /src/install.r
```
options(repos = list(CRAN = "http://cran.rstudio.com/"))
```
which overwrites my personal [CRAN] mirror. 
Tbh as mentioned above I don't know if this more of a feature request than a "real" issue?    
I tried the "easy" or some form of workaround ie forking this repo, changed that line 8 in `install.r` and adjust the masons extensions `registry.json` file line xxx to point to that personal fork (during the install routine, sort of), but any changes to that `registry.json` file will get overwritten prior to/during the `LspInstall` process :-( 
And I am not familiar enough with the mason or lsp extensions inner workings to dig deeper into further workarounds or changes to their lsp installation processes. 
Therefore it would be really helpful if there was a way to avoid forcing that 'cran.rstudio.com' repo into the package installation process.   
Alternatively, maybe there's a setting somewhere to use an already installed instance of `library(languageserver)`, instead of triggering a full "new install" of the `languageserver` package, which I just have not been able to find (yet) somewhere eg in the documentation?    
Also no problem to go down that latter route - if someone can point me in the right direction?

### PR DESCRIPTION
**Pull request by williamboman**
Also change the default repo URL to https://cloud.r-project.org.

Fixes #2.

### CODE REVIEW

File: src/install.r [hunk 1]
```diff
@@ -5,7 +5,11 @@ ref <- args[2]
 
 options(langserver_library = langserver_library)
 options(langserver_quiet = FALSE)
-options(repos = list(CRAN = "http://cran.rstudio.com/"))
+local({
+  current_cran <- getOption("repos")["CRAN"]
+  if (current_cran == "@CRAN@")
```
Comment (williamboman) [inline src/install.r:10]: I don't particularly like checking for the "magic" value `@CRAN@` but the docs seem to suggest it's fine:

```
Initially set from entries marked as default in the ‘repositories’ file, see
R_REPOSITORIES.  The ‘factory-fresh’ setting from ‘etc/repositories’ is
‘c(CRAN="@CRAN@")’, a value that causes some utilities to prompt for a CRAN
mirror.
```

Comment (gwd666) [reply src/install.r:10]: I am ok with that ... maybe @harmongrey has additional comments?

Comment (harmongray) [reply src/install.r:10]: Hi - sorry about the delay. I definitely think this is the most effective implementation, and good check on the `@CRAN@` value, as that could cause funky issues. Thanks for this small, but useful fix!

**Review by harmongray — APPROVED** [review]

**Review by gwd666 — APPROVED** [review]
Thnx a lot - also like that you thought about the switch to cloud.r-project.org which afaik will try to reach the "closest" mirror (maintained by the 'official' R crew and no longer relying on rstudio's  mirror in there). 
As soon as this is merged I will test it by running the r-langserver MasonInstall behind my proxy and let you know if it succeeds now.

**Pull request merged**